src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -53,9 +51,59 @@ class HttpGoogleCloudStorageDAO(
 
   val system = akka.actor.ActorSystem("system")
 
-  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String): Unit = {
-    val inserter = getStorage(getBucketCredential(userInfo)).buckets.insert(projectId, new Bucket().setName(getBucketName(workspaceId)))
-    retry(inserter.execute,when500)
+  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String, workspaceName: WorkspaceName): Unit = {
+    val bucketName = getBucketName(workspaceId)
+    val groups = getGroupDirectory.groups
+    insertGroup(groups,newGroup(bucketName,workspaceName,WorkspaceAccessLevel.Read)) {
+      insertGroup(groups,newGroup(bucketName,workspaceName,WorkspaceAccessLevel.Write)) {
+        insertGroup(groups,newGroup(bucketName,workspaceName,WorkspaceAccessLevel.Owner)) {
+          val readersEntity = makeGroupEntityString(toGroupId(bucketName,WorkspaceAccessLevel.Read))
+          val writersEntity = makeGroupEntityString(toGroupId(bucketName,WorkspaceAccessLevel.Write))
+          val ownersEntity = makeGroupEntityString(toGroupId(bucketName,WorkspaceAccessLevel.Owner))
+          val userEntity = s"user-${userInfo.userEmail}"
+          val bucket = new Bucket().setName(bucketName).setAcl(List(
+              newBucketAccessControl(readersEntity,WorkspaceAccessLevel.Read),
+              newBucketAccessControl(writersEntity,WorkspaceAccessLevel.Write),
+              newBucketAccessControl(ownersEntity,WorkspaceAccessLevel.Owner),
+              newBucketAccessControl(userEntity,WorkspaceAccessLevel.Owner)
```

Comment:
yes, i figured that would be controversial.  it seems to me that this solves a bunch of problems that can occur when a user accidentally demotes him or herself from owner status.  the owner is always the owner.  could be a nice feature.
that being said, i'll remove it.